File: exercises/D205/solution.pas
var
    n, m, i: longint;
    first: boolean;
begin
    read(n);
    m := trunc(sqrt(n));
    first := true;
    write(n, '=');
    for i := 2 to m do
        while (n mod i = 0) do
        begin
            if (not first) then
                write('*');
            write(i);
            n := n div i;
            first := false;
        end;
    if (n > 1) then
    begin
        if (not first) then
            write('*');
        write(n)
    end;
    writeln
end.

File: exercises/D205/solution.cpp
#include <cmath>
#include <iostream> 
using namespace std;
int main() {
    int n;
    cin >> n;
    int m = sqrt(n);
    bool first = true;
    cout << n << "=";
    for (int i = 2; i <= m; i++) {
        while (n % i == 0) {
            if (!first) {
                cout << "*";
            }
            cout << i;
            n /= i;
            first = false;
        }
    }
    if (n > 1) {
        if (!first) {
            cout << "*";
        }
        cout << n;
    }
    cout << endl;
}
